### ISSUE TITLE
Codefix doesn’t respect suppression.

### ISSUE THREAD

**Issue by jinujoseph** [dotnet/roslyn#2769]
![ignoring supression](https://cloud.githubusercontent.com/assets/12449387/7643223/511b5e6c-fa4c-11e4-8f91-b65b4e02890a.png)

**Comment by sharwell** [comment]
The problem here is the **Remove Unnecessary Usings** code fix should actually be **Remove Unnecessary Using**, and allow **Fix all occurrences in: Document** to be used to handle the set. This *would* respect suppressions.

On a separate note, why would you need this suppression? For both cases I can think of, it seems like there are better ways to approach it:

1. **Documentation comments reference an item:** If you have documentation comments with `<see cref="List{T}"/>` (for example), and you are still having the IDE suggest you remove the using statement for `System.Collections.Generic`, it means you are not generating documentation comment files for your project. When documentation is not being generated, the contents of the XML comments are *completely ignored* by the compiler for all operations. I just migrated an old CodePlex issue related to this subject to #2773.

2. **Items shared across multiple projects:** In this case, make sure to use the same preprocessor symbols for your `using` declarations as you use for the code which needs them. This way, regardless of which project context you are in the result of the operation is valid.

**Comment by shyamnamboodiripad** [comment]
Yes I think this is a side effect of the fact that the **remove unused usings** fix is implemented via multiple diagnostics. IIRC, the first is produced by the compiler and spans exactly a single unused usings, another (i.e. IDE0005) coalesces the individual but contiguous unused using diagnostics into a single contiguous one and yet another produces a single diagnostic that spans the entire block of usings (both used and unused). Light bulb is triggered off of the last one so that one doesn't have to be on the exact line that contains an unused using in order to trigger the fix. I agree implementing this via Fix all occurrences in: Document may probably be better and would avoid all this complexity - the only down side would be that we can't offer the light bulb for this fix on a line that contains a used using. 

Now, I suspect this bug is happening because the IDE0005 diagnostic (that is being suppressed above and that is one of the diagnostics I mentioned above) has been changed to 'warning' (from 'hidden') but the other diagnostics involved are still hidden. In other words, we probably wouldn't run into this if we hadn't changed this severity :)
